pkg/types/baremetal/platform.go

```diff
@@ -12,15 +12,17 @@ type BMC struct {
 	DisableCertificateVerification bool   `json:"disableCertificateVerification"`
 }
 
-// BootMode puts the server in legacy (BIOS) or UEFI mode for
-// booting. The default is UEFI.
-// +kubebuilder:validation:Enum=UEFI;legacy
+// BootMode puts the server in legacy (BIOS), UEFI secure boot or UEFI mode for
+// booting. Secure boot is only enabled during the final instance boot.
+// The default is UEFI.
+// +kubebuilder:validation:Enum=UEFI;UEFISecureBoot;legacy
```

Comment:
Shouldn't we also have "" as a valid value?
```suggestion
// +kubebuilder:validation:Enum="";UEFI;UEFISecureBoot;legacy
```

Reply:
Probably, but this could get fixed separately.  I think I'd also want pkg/types/baremetal/defaults to set UEFI on the host. Today it's just done behind the scenes in the tfvars

Reply:
Okay, I won't change it here then.

Reply:
https://github.com/openshift/installer/pull/4680